### ISSUE TITLE
Loop Controls

### ISSUE THREAD

**Issue by vrtc** [mozilla/nunjucks#296]
There is no possibility to stop the cycle, such as:

``` html
{% for user in users %}
    {%- if loop.index >= 10 %}{% break %}{% endif %}
{%- endfor %}
```

it returns an error:
unknown block tag: break

**Comment by ArmorDarks** [comment]
it would be really nice to see `break` tag since it the only way to implement `while` loops in Jinja and nunjucks

**Comment by roll** [comment]
:+1:

**Comment by wjthomas9** [comment]
Just stumbled across this looking for the same feature. This would be a welcome feature.

**Comment by ahoef** [comment]
Agreed, this would be a useful feature!

**Comment by felipenmoura** [comment]
yep...
:+1:

**Comment by j4in** [comment]
:+1:

**Comment by jinmatt** [comment]
:+1:  definitely useful feature

**Comment by kamlekar** [comment]
I think this is not an enhancement but a bug.

**Comment by kruncher** [comment]
:+1:

**Comment by ryanvanderpol** [comment]
+1

**Comment by rotexhawk** [comment]
+1

**Comment by samuells** [comment]
+1

**Comment by balazj** [comment]
Next/Continue would also be nice. Currently same thing can be implemented with IF conditions, but this would produce more readable code.

**Comment by blakeperdue** [comment]
A year later, still would love a `break` :)

**Comment by AlJohri** [comment]
Would also love to see `continue` implemented!

http://jinja.pocoo.org/docs/2.10/extensions/#loop-controls
http://jinja.pocoo.org/docs/2.10/templates/#loop-controls

https://github.com/pallets/jinja/blob/7a6704db55fcd9bbe5a90c3868e03f5a5fcf176a/jinja2/ext.py#L418-L426
https://github.com/pallets/jinja/blob/7a6704db55fcd9bbe5a90c3868e03f5a5fcf176a/jinja2/nodes.py#L948-L953

**Comment by maddog986** [comment]
I also would love to see this.

**Comment by lahloug** [comment]
Break and continue has been implemented as an extension called `loopcontrols`:

I added this line on my flask app and it worked fine !

`app.jinja_env.add_extension('jinja2.ext.loopcontrols')`

**Comment by bzeklaf** [comment]
Hi guys, any idea how i can actually implement loopextensions?

**Comment by narenaryan** [comment]
I think one should activate Jinja2 `loopcontrols` extension. Otherwise, Jinja won't recognize a
 `{% break %}`

For Example:
```python
from jinja2 import Environment

env = Environment(
        extensions=["jinja2.ext.loopcontrols"]
)

content = '<YOUR TEMPLATE STRING_WITH_BREAK>'
template = env.from_string(content)
print(template.render())
```
For more information see this link:
https://jinja.palletsprojects.com/en/2.10.x/templates/#loop-controls

**Comment by lucasdavidferrero** [comment]
This will be really powerful!

**Comment by AdryAtDev** [comment]
Still waiting!

**Comment by viharm** [comment]
Yes, I'd love to know how this is done too. Otherwise the workaround is to have a local in-loop boolean, which is flipped; then that is used to skip through further iterations, but it is far from elegant.

**Comment by andreas-prouza** [comment]
2024: Also agree for break & continue :+1:
